syncweek-react-aad/react-aad
# Fix unauthenticated callback being called even after login

## Purpose
FIxes issue after login where the check for isLoggedIn() executes before MSAL manages to save the user information into storage.

## Does this introduce a breaking change?

```
[ ] Yes
[n ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  In the SampleAppRedirectOnLaunch.js file, add `console.log` statements in the `unathenticatedFunction`, `authenticatedFunction`, and the `userJustLoggedIn`.
* After login, you should see the order of callbacks executed as such: `authenticatedFunction` => `userJustLoggedIn`. At no point should the `unathenticatedFunction` function execute.
* While logged in, refreshing the page should execute the same callbacks in similar order. `unathenticatedFunction` should never execute.
* After logging out, only the `unathenticatedFunction` should execute.

## What to Check
* I saw some code in the `MsalRedirectAuthProvider` which never seemed to execute. I'm not sure what the original purpose of it was, so I removed it. If possible, it would be helpful for someone to give a cursory look over that to make sure I haven't removed something meant to resolve an edge case somewhere.

## Other Information
* To resolve this, I moved much of the logic that set the `AuthenticationState` enum into the `MsalAuthProvider` where we have the closest access to the MSAL code itself. When the state changes, it executes a callback which will pass the information into the `AzureAD`. This modifies the component from a polling pattern into more of a subscribe.
